### PR TITLE
slackpkg: fix matching some special cases in package names.

### PR DESCRIPTION
**Pull request by majekw**
##### SUMMARY

There are some special cases in Slackware package naming which are not covered by `$ARCH|noarch` condition:
- package `kernel-headers` has architecture set to `x86` on `x86_64`
- there are also some firmware packages with architecture set to `fw`

This PR adds check for `kernel-headers` special case and adds `fw` to other expected architectures.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

slackpkg.py

##### ADDITIONAL INFORMATION

For example, try to install package `kernel-headers`.
Module runs `slackpkg install kernel-headers` (installs it if package is not present), then check is run if package is installed, but it can't find it because architecture part of package name is different (`x86` instead of expected `x86_64`) and task fails.

Similar thing for firmware files, for example `zd1211-firmware-1.4-fw-1`. It fails because architecture part is set to `fw` and doesn't match current condition.

Example output from system where `kernel-headers` are already installed
```paste below
$ansible mii -m slackpkg -a "name=kernel-headers"
mii | FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "msg": "failed to install kernel-headers: \nLooking for kernel-headers in package list. Please wait... DONE\n\nNo packages match the pattern for install. Try:\n\n\t/usr/sbin/slackpkg reinstall|upgrade \n\n\n",
    "stderr": "",
    "stderr_lines": []
}
```

### CODE REVIEW

**Comment by felixfontein** [comment]
Looks good to me. @KimNorgaard could you please take a look?

File: changelogs/fragments/505-slackpkg_fix_matching_some_special_cases_in_package_names.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+ - slackpkg - fix matching some special cases in package names.
```
Comment (Andersson007) [inline changelogs/fragments/505-slackpkg_fix_matching_some_special_cases_in_package_names.yml:2]:
```suggestion
 - slackpkg - fix matching some special cases in package names (https://github.com/ansible-collections/community.general/pull/505).
```

**Comment by felixfontein** [comment]
@majekw do you want to continue to work on this, or should we merge it?

**Comment by majekw** [comment]
It's ready to merge.
I force pushed as Github choose wrong e-mail for commit, so I corrected it, but nothing more was changed. Sorry for making confusion.

**Comment by felixfontein** [comment]
I'll wait a bit more so that @KimNorgaard can take a look when possible.

**Comment by KimNorgaard** [comment]
Looks good

**Comment by felixfontein** [comment]
@majekw @Andersson007 @KimNorgaard thanks a lot!

**Pull request merged**